src/extension/ai/converters.ts

```diff
@@ -0,0 +1,86 @@
+// Conversion routines between the protocol buffer format and VSCode's representation of a notebook
+//
+// See ../vscode_apis.md for an exlanation. It is very helpful for understanding this folder.
+
+import * as vscode from 'vscode'
+//import * as metadata from './metadata'
+//import * as constants from './constants'
+import {
+  Notebook,
+  Cell,
+  CellKind,
+  CellOutput,
+  CellOutputItem,
+} from '@buf/stateful_runme.community_timostamm-protobuf-ts/runme/parser/v1/parser_pb'
+
+// cellToCellData converts a NotebookCell to a NotebookCellData.
+// NotebookCell is an interface used by the editor.
+// NotebookCellData is a concrete class.
+export function cellToCellData(cell: vscode.NotebookCell): vscode.NotebookCellData {
```

Comment:
@sourishkrout Do I need these conversion routines to convert between the protos and vscode's native data structures? Are there existing routines I can reuse?

Reply:
this static method should do the job: https://github.com/stateful/vscode-runme/blob/main/src/extension/serializer.ts#L302-L333

Reply:
@jlewi is there maybe a file missing in the latest commits?

<img width="586" alt="image" src="https://github.com/stateful/vscode-runme/assets/250527/ba152e58-e314-448b-bf37-cff5a4c9c580">

Reply:
@sourishkrout Can you help me figure out what's going on here and how to fix it.  I think it should have been

```
import {Serializer} from '../../types'
```

I'm trying to import [Notebook defined here](https://github.com/stateful/vscode-runme/blob/497053bb5b147b0ceb4a1d9b9cc62c925cd66543/src/types.ts#L63).

I'm trying to instantiate an instance of Notebook to pass to Notebook.Revive here
https://github.com/stateful/vscode-runme/pull/1356/files#diff-7561d1e499859730c4937494e85d7dddfbe11332a496476afc5d082860df7335R81

What's the proper way

1. Instantiate a notebook
2. Populate the cells in the notebook with the proto

I'm a little confused by what's going on. I see the error in vscode but ` npm run bundle` succeeds and I thought I had installed it successfully from source.

Reply:
So I think I partially understand the error. Serializer.Notebook is a type not a class so I can't use new.  I wrote some conversion code to convert the proto type to the Serializer.CellType

PTAL

Reply:
Correct. The `Serializer.Notebook` is a type that in structural typing sense fits different Parsers. Runme has a legacy WASM-based parser from its prototype days. Fixing the import path fixed the error for me.

I have seen the Typescript LSP and/or the ESlint LSP getting "confused" which happens when files are being moved around right under it. The easy fix is to restart them using the command palette. In any case, it compiles cleanly now on all fronts webpack/tsc, and eslint & typescript LSPs.